pkg/security/oauth2/oauth/oauth.go

```diff
@@ -57,6 +64,13 @@ func (c *Config) Validate() error {
 	if c.ClientSecretFile == "" {
 		return fmt.Errorf("clientSecretFile('%v')", c.ClientSecretFile)
 	}
+	switch c.GrantType {
+	case ClientCredentials, AuthorizationCode:
```

Comment:
shouldn't we validate that ownerId is set if at least one client creds provider is configured on the coap-gateway? Reason is that if it comes to this combination, no ownerId set - sub is used, and client credential is used, super-device is created - it can query all devices. Can it?